src/training/tech.rs

```diff
@@ -1,278 +1,269 @@
+    status_kind: L2CValue,
+    unk: L2CValue,
+) -> L2CValue {
+    let mut status_kind = status_kind;
+    let mut unk = unk;
+
+    if is_training_mode() {
+        mod_handle_change_status(fighter, &mut status_kind, &mut unk);
+    }
+
+    original!()(fighter, status_kind, unk)
+}
+
+unsafe fn mod_handle_change_status(
+    fighter: &mut L2CFighterBase,
+    status_kind: &mut L2CValue,
+    unk: &mut L2CValue,
+) {
+    let module_accessor = sv_system::battle_object_module_accessor(fighter.lua_state_agent);
+    if !is_operation_cpu(module_accessor) {
+        return;
+    }
+
+    let status_kind_int = status_kind
+        .try_get_int()
+        .unwrap_or(*FIGHTER_STATUS_KIND_WAIT as u64) as i32;
+
+    let state: TechFlags = MENU.tech_state.get_random();
+
+    if handle_grnd_tech(module_accessor, status_kind, unk, status_kind_int, state) {
+        return;
+    }
+
+    if handle_wall_tech(module_accessor, status_kind, unk, status_kind_int, state) {
+        return;
+    }
+
+    handle_ceil_tech(module_accessor, status_kind, unk, status_kind_int, state);
+}
+
+unsafe fn handle_grnd_tech(
+    module_accessor: &mut app::BattleObjectModuleAccessor,
+    status_kind: &mut L2CValue,
+    unk: &mut L2CValue,
+    status_kind_int: i32,
+    state: TechFlags,
+) -> bool {
+    if status_kind_int != *FIGHTER_STATUS_KIND_DOWN
+        && status_kind_int != *FIGHTER_STATUS_KIND_DAMAGE_FLY_REFLECT_D
+    {
+        return false;
+    }
+
+    // prev_status_kind(module_accessor, 0) gets the 1st previous status,
+    // which is FIGHTER_STATUS_KIND_CATCHED_AIR_END_GANON for both aerial/grounded sideb
+    // prev_status_kind(module_accessor, 1) gets the 2nd previous status,
+    // which is FIGHTER_STATUS_KIND_CATCHED_GANON for grounded sideb
+    // and FIGHTER_STATUS_KIND_CATCHED_AIR_GANON for aerial sideb
+    let second_prev_status = StatusModule::prev_status_kind(module_accessor, 1);
+    let can_tech = WorkModule::is_enable_transition_term(
+        module_accessor,
+        *FIGHTER_STATUS_TRANSITION_TERM_ID_PASSIVE,
+    ) && (second_prev_status != FIGHTER_STATUS_KIND_CATCHED_AIR_FALL_GANON);
+
+    if !can_tech {
+        return false;
+    }
+
+    let do_tech: bool = match state {
+        TechFlags::IN_PLACE => {
+            *status_kind = FIGHTER_STATUS_KIND_PASSIVE.as_lua_int();
+            *unk = LUA_TRUE;
+            true
+        }
+        TechFlags::ROLL_F => {
+            *status_kind = FIGHTER_STATUS_KIND_PASSIVE_FB.as_lua_int();
+            *unk = LUA_TRUE;
+            TECH_ROLL_DIRECTION = Direction::IN; // = In
+            true
+        }
+        TechFlags::ROLL_B => {
+            *status_kind = FIGHTER_STATUS_KIND_PASSIVE_FB.as_lua_int();
+            *unk = LUA_TRUE;
+            TECH_ROLL_DIRECTION = Direction::OUT; // = Away
+            true
+        }
+        _ => false,
+    };
+    if do_tech && MENU.mash_triggers.contains(MashTrigger::TECH) {
+        mash::buffer_menu_mash();
+    }
+
+    true
```

Comment:
Should this be `do_tech`?

Reply:
No, it should stay as true (I didn't change this logic). The return value of these functions represents whether the CPU **could have** teched in this manner at all, not whether they **did** tech. The return value is used to skip the remaining tech scenario checks; if the CPU could have performed a ground tech, you know they aren't in a wall or ceiling tech scenario. 

That said, I agree that it's confusing. The checks are inexpensive so we can remove this logic if you'd prefer.